File: noti-send/src/main/java/com/ezbuy/notisend/service/MailService.java
package com.ezbuy.notisend.service;

import com.ezbuy.notimodel.dto.EmailResultDTO;
import com.ezbuy.notimodel.dto.TransmissionNotiDTO;
import java.util.List;
import reactor.core.publisher.Mono;

public interface MailService {
    Mono<List<EmailResultDTO>> sendMailByTransmission(List<TransmissionNotiDTO> transmissionNotis);
}

File: notification-send/src/main/kotlin/com/ezbuy/notificationsend/service/MailService.kt
package com.ezbuy.notificationsend.service

import com.ezbuy.notimodel.dto.EmailResultDTO
import com.ezbuy.notimodel.dto.TransmissionNotiDTO
import reactor.core.publisher.Mono

interface MailService {
    fun sendMailByTransmission(transmissionNotis: List<TransmissionNotiDTO>): Mono<List<EmailResultDTO>>
}